db/version_set.cc

```diff
@@ -1239,66 +1248,42 @@ VersionSet::UpdatePenalty(
                         increment=8;
                     }   // if
                     else
-                    {   // geometric penalty
-                        value=count;
-                        increment=2;
+                    {   // slightly less penalty
+                        value=count+1;
+                        count=0;
                     }   // else
-
-                    for (loop=0; loop<count; ++loop)
-                        value*=increment;
-
-                    penalty+=value;
                 }   // else
             }   // if
         }   // if
         else
         {
-#if 0
-            // Compute the ratio of current size to size limit.
-            double penalty_score(0.0);
-
-            const uint64_t level_bytes = TotalFileSize(v->files_[level]);
-
-	    // cubed penalty for exceeding size of sorted level
-	    //  (especially helps tiered storage)
-            penalty_score = static_cast<double>(level_bytes) / gLevelTraits[level].m_MaxBytesForLevel;
-	    penalty_score = penalty_score * penalty_score * penalty_score;
-
-	    // if no penalty so far, set a minor penalty to the landing
-	    //   level to help it flush.  because first sorted layer needs to 
-	    //   clear before next dump of overlapped files.
-	    if (penalty_score<1.0 && config::kNumOverlapLevels==level)
-            {
-                penalty_score = (1.0<(static_cast<double>(level_bytes) / gLevelTraits[level].m_DesiredBytesForLevel)? 1.0 : 0.0);
-            }   // if
-
-            if (1.0<=penalty_score)
-                penalty+=(static_cast<int>(penalty_score));
-#else
-	    int loop, count, value, increment;
             const uint64_t level_bytes = TotalFileSize(v->files_[level]);
 
 	    count=static_cast<double>(level_bytes) / gLevelTraits[level].m_MaxBytesForLevel;
```

Comment:
corrected.

> On Aug 9, 2015, at 10:05 PM, Paul A. Place <notifications@github.com> wrote:
> 
> In db/version_set.cc <https://github.com/basho/leveldb/pull/162#discussion_r36597806>:
> 
> > -	    //  (especially helps tiered storage)
> > -            penalty_score = static_cast<double>(level_bytes) / gLevelTraits[level].m_MaxBytesForLevel;
> > -	    penalty_score = penalty_score * penalty_score * penalty_score;
> > -
> > -	    // if no penalty so far, set a minor penalty to the landing
> > -	    //   level to help it flush.  because first sorted layer needs to 
> > -	    //   clear before next dump of overlapped files.
> > -	    if (penalty_score<1.0 && config::kNumOverlapLevels==level)
> > -            {
> > -                penalty_score = (1.0<(static_cast<double>(level_bytes) / gLevelTraits[level].m_DesiredBytesForLevel)? 1.0 : 0.0);
> > -            }   // if
> > -
> > -            if (1.0<=penalty_score)
> > -                penalty+=(static_cast<int>(penalty_score));
> > -#else
> > -	    int loop, count, value, increment;
> >              const uint64_t level_bytes = TotalFileSize(v->files_[level]);
> >  
> >  	    count=static_cast<double>(level_bytes) / gLevelTraits[level].m_MaxBytesForLevel;
> Have tabs instead of spaces in the following lines: 1262, 1264, 1266, 1858, 1859, 1861
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/basho/leveldb/pull/162/files#r36597806>.
>